### DISCUSSION TITLE
[2기-B] 윤이나

### DISCUSSION THREAD

**Discussion by devmei** [super-corini/programmers_kdt_II#360] [2기 Monthly Project I]
## [Monthly Project I] 데이터 시각화 웹 페이지 만들기
[Students Performance in Exams](https://www.kaggle.com/spscientist/students-performance-in-exams) 데이터에 대한 EDA
* **웹 사이트** → http://devmei.pythonanywhere.com/EDA/boot
* **git** → https://github.com/devmei/programmers_kdt_II/tree/week5_youn/MonthlyProject/projects/mysite
---
* `AWS` 프리 티어 사용량을 초과하여 `PythonAnywhere` 사용하였습니다.
  * 배포가 정상적으로 되지 않아 시간이 꽤 걸렸는데, 전부 삭제하고 차근차근하니 해결되었습니다. 😇
  * 추후 `GCP` 사용해 보겠습니다.
* Bonus Part의 **웹페이지 상에서 코드의 parameter를 조정할 수 있는 기능**은
  **인터랙티브 그래프**(_Interactive Graph_)로 이해하고 `plotly`를 사용해 그래프를 그렸습니다.
  * 그래프를 `to_json()`으로 `json` 형식으로 변환한 후 웹 페이지에 첨부하였습니다.
  * 데이터의 양이 많아 `static` 폴더 내에 `graph` 폴더를 생성하여 그래프의 `.json` 파일을 모아두었습니다.
    ``` .js
    { json 정보를 담을 변수 이름 } = { `to_json()`으로 변환한 데이터 }
    ```
  * `.html` 파일에 `plotly.js`를 추가합니다.
    ``` .html
    <!-- Plotly -->
    <script src="https://cdn.plot.ly/plotly-latest.min.js"></script>
    ```
  * 그래프를 넣고 싶은 위치에 아래 코드를 작성합니다.
    ```.html
    <div id="{ <div> ID(이름) }">
        <script type="text/javascript" src="{ .json 파일 경로 }"></script>
        <script>Plotly.plot('{ <div> ID(이름) }', { json 정보를 담은 변수 이름 }, {});</script>
    </div>
    ```
* `Bootstrap` _Free Template_ 적용하였습니다.

**Comment by super-corini** [comment]
깔끔합니당. 👍 
다음 배포땐 시간이 절반 이상 줄어드실 것이라 자부합니당. 수고하셨어요!
